Some deprecation warnings during compile

Just had a quick try and everything seems to work fine so far. Didn't get into the nitty-gritty details like context menu workspace size constraining and similar yet.

This is just a compile time warning but I thought it might be worth to point out:

```
[12/26] Compiling C object examples/gtk4-layer-demo.p/demo_gtk-layer-demo.c.o
../examples/demo/gtk-layer-demo.c: In function ‘main’:
../examples/demo/gtk-layer-demo.c:422:5: warning: ‘G_APPLICATION_FLAGS_NONE’ is deprecated: Use 'G_APPLICATION_DEFAULT_FLAGS' instead [-Wdeprecated-declarations]
  422 |     GtkApplication * app = gtk_application_new ("com.github.wmww.gtk4-layer-shell.demo", G_APPLICATION_FLAGS_NONE);
      |     ^~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/gio/giotypes.h:30,
                 from /usr/include/glib-2.0/gio/gio.h:28,
                 from /usr/include/gtk-4.0/gtk/css/gtkcsssection.h:25,
                 from /usr/include/gtk-4.0/gtk/css/gtkcss.h:37,
                 from /usr/include/gtk-4.0/gtk/gtk.h:30,
                 from ../include/gtk4-layer-shell.h:4,
                 from ../examples/demo/gtk-layer-demo.h:4,
                 from ../examples/demo/gtk-layer-demo.c:1:
/usr/include/glib-2.0/gio/gioenums.h:1536:3: note: declared here
 1536 |   G_APPLICATION_FLAGS_NONE GLIB_DEPRECATED_ENUMERATOR_IN_2_74_FOR(G_APPLICATION_DEFAULT_FLAGS),
      |   ^~~~~~~~~~~~~~~~~~~~~~~~
[22/26] Compiling C object examples/simple-example-c.p/simple-example.c.o
../examples/simple-example.c: In function ‘main’:
../examples/simple-example.c:51:5: warning: ‘G_APPLICATION_FLAGS_NONE’ is deprecated: Use 'G_APPLICATION_DEFAULT_FLAGS' instead [-Wdeprecated-declarations]
   51 |     GtkApplication * app = gtk_application_new ("com.github.wmww.gtk4-layer-shell.example", G_APPLICATION_FLAGS_NONE);
      |     ^~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/gio/giotypes.h:30,
                 from /usr/include/glib-2.0/gio/gio.h:28,
                 from /usr/include/gtk-4.0/gtk/css/gtkcsssection.h:25,
                 from /usr/include/gtk-4.0/gtk/css/gtkcss.h:37,
                 from /usr/include/gtk-4.0/gtk/gtk.h:30,
                 from ../include/gtk4-layer-shell.h:4,
                 from ../examples/simple-example.c:1:
/usr/include/glib-2.0/gio/gioenums.h:1536:3: note: declared here
 1536 |   G_APPLICATION_FLAGS_NONE GLIB_DEPRECATED_ENUMERATOR_IN_2_74_FOR(G_APPLICATION_DEFAULT_FLAGS),
      |   ^~~~~~~~~~~~~~~~~~~~~~~~
```

And one more from meson itself:
```
NOTICE: Future-deprecated features used:
 * 0.56.0: {'dependency.get_pkgconfig_variable'
```

Another one from wayland-scanner (I suppose):
```
[2/26] Generating 'src/libgtk4-layer-shell.so.1.0.0.p/wlr-layer-shell-unstable-v1.c'.
Using "code" is deprecated - use private-code or public-code.
See the help page for details.
[3/26] Generating 'src/libgtk4-layer-shell.so.1.0.0.p/xdg-shell.c'.
Using "code" is deprecated - use private-code or public-code.
See the help page for details
```

Edit:
> Didn't get into the nitty-gritty details like context menu workspace size constraining and similar yet.

This seems fixed, at least dropdowns and popover menus now end at the bottom of the screen and allow scrolling to reach the remaining items.

--- a/doc/meson.build
+++ b/doc/meson.build
@@ -1,6 +1,6 @@
 subdir('xml')
 
-glib_prefix = dependency('glib-2.0').get_pkgconfig_variable('prefix')
+glib_prefix = dependency('glib-2.0').get_variable(pkgconfig: 'prefix')
 glib_docpath = join_paths(glib_prefix, 'share', 'gtk-doc', 'html')
 docpath = join_paths(get_option('datadir'), 'gtk-doc', 'html')
 

--- a/meson.build
+++ b/meson.build
@@ -2,7 +2,7 @@ project('gtk4-layer-shell',
     ['c'],
     version: '1.0.0',
     license: 'MIT',
-    meson_version: '>=0.45.1',
+    meson_version: '>=0.51.0',
     default_options: ['c_std=gnu11', 'warning_level=3'])
 
 lib_so_version = '0'

--- a/protocol/meson.build
+++ b/protocol/meson.build
@@ -28,7 +28,6 @@ gen_server_header = generator(prog_wayland_scanner,
 gen_private_code = generator(prog_wayland_scanner,
     output: ['@BASENAME@.c'],
     arguments: ['-c', 'private-code', '@INPUT@', '@BUILD_DIR@/@BASENAME@.c'])
-# 'code' is deprecated, and can be replaced with 'private-code' when all platforms have a new enough wayland-scanner
 
 client_protocol_srcs = []
 server_protocol_srcs = []

--- a/protocol/meson.build
+++ b/protocol/meson.build
@@ -27,7 +27,7 @@ gen_server_header = generator(prog_wayland_scanner,
 
 gen_private_code = generator(prog_wayland_scanner,
     output: ['@BASENAME@.c'],
-    arguments: ['-c', 'code', '@INPUT@', '@BUILD_DIR@/@BASENAME@.c'])
+    arguments: ['-c', 'private-code', '@INPUT@', '@BUILD_DIR@/@BASENAME@.c'])
 # 'code' is deprecated, and can be replaced with 'private-code' when all platforms have a new enough wayland-scanner
 
 client_protocol_srcs = []

--- a/protocol/meson.build
+++ b/protocol/meson.build
@@ -1,5 +1,5 @@
 if wayland_scanner.found()
-    prog_wayland_scanner = find_program(wayland_scanner.get_pkgconfig_variable('wayland_scanner'))
+    prog_wayland_scanner = find_program(wayland_scanner.get_variable(pkgconfig: 'wayland_scanner'))
 else
     prog_wayland_scanner = find_program('wayland-scanner')
 endif
@@ -10,7 +10,7 @@ protocols = [
 
 if wayland_protocols.found()
     protocols += join_paths(
-        wayland_protocols.get_pkgconfig_variable('pkgdatadir'),
+        wayland_protocols.get_variable(pkgconfig: 'pkgdatadir'),
         'stable/xdg-shell/xdg-shell.xml')
 else
     # use bundled xdg-shell.xml

